pages/vault.js

```diff
@@ -136,19 +160,22 @@ export default function Vault () {
         const _options = { from: account }
 
         try {
+          const _filesCount = await contract.getFilesCount(_options)
+          setTotalFiles(prevState => _filesCount)
+          setTotalPages(prevState => Math.ceil(_filesCount / pageSize))
+
+          console.log(firstIndex, lastIndex)
           console.log('Fetching files metadata form blockchain...')
-          const _files = await contract.getFiles(_options)
-          if (_files.length > 0) {
-            setFiles(prevState => _files)
-          }
+          const _files = await contract.getRangeOfFiles(firstIndex, lastIndex, _options)
+          setFiles(prevState => _files)
         } catch (err) {
           console.log('Cannot fetch files from blockchain:', err.message)
           // TODO: handle error
         }
       }
     }
     fetchFilesMetadata()
-  }, [isTransactionSucceed, chainId, account, contract, isAuthorized])
+  }, [isTransactionSucceed, chainId, account, contract, isAuthorized, firstIndex, lastIndex, currentPage])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'pageSize'. Either include it or remove the dependency array. You can also replace multiple useState variables with useReducer if 'setTotalPages' needs the current value of 'pageSize'.](https://app.codacy.com/gh/AbdeltwabMF/devault/pullRequest?prid=9759674)
```suggestion
  }, [isTransactionSucceed, chainId, account, contract, isAuthorized, firstIndex, lastIndex, currentPage, pageSize])
```